### PR TITLE
Fix sorting of simulation results using revision number

### PR DESCRIPTION
**Pull request by hoanphungt**

### CODE REVIEW

File: gui/threedi_plugin_grid_result_dialog.py [hunk 1]
```diff
@@ -209,24 +210,31 @@ def refresh(self):
...
                     result_item.setData(os.path.join(local_revision.results_dir, result_dir))
-                    self.model.appendRow([schema_item, revision_item, result_item])
+                    rows.append([schema_item, revision_item, result_item])
 
                 # In case no results are present, but a gridadmin is present, we still add the grid, but without result item
                 if num_of_results == 0 and os.path.exists(os.path.join(local_revision.grid_dir, "gridadmin.h5")):
                     schema_item = QStandardItem(local_schematisation.name)
                     schema_item.setEditable(False)
                     revision_item = QStandardItem(str(revision_number))
+                    revision_item.setData(revision_number, Qt.UserRole)
```
Comment (ldebek) [inline gui/threedi_plugin_grid_result_dialog.py]: shouldn't it be a `Qt.DisplayRole` ?

File: gui/threedi_plugin_grid_result_dialog.py [hunk 2]
```diff
@@ -209,24 +210,31 @@ def refresh(self):
...
```
Comment (ldebek) [inline gui/threedi_plugin_grid_result_dialog.py]: shouldn't it be a `Qt.DisplayRole` ?

File: gui/threedi_plugin_grid_result_dialog.py [hunk 3]
```diff
@@ -209,24 +210,31 @@ def refresh(self):
...
                     revision_item.setEditable(False)
                     # We'll store the grid folder with the revision item for fast retrieval
                     revision_item.setData(local_revision.grid_dir)
-                    self.model.appendRow([schema_item, revision_item])
+                    rows.append([schema_item, revision_item])
+
+        # Sort table rows by revision number using the UserRole
+        rows.sort(key=lambda x: x[1].data(Qt.UserRole))
```
Comment (ldebek) [inline gui/threedi_plugin_grid_result_dialog.py]: ...and `Qt.DisplayRole` as well?

File: gui/threedi_plugin_grid_result_dialog.py (hunk 1 above)
Comment (hoanphungt) [reply gui/threedi_plugin_grid_result_dialog.py]: It didn't work for me at first using DisplayRole so I switched to UserRole but it works now. Maybe I got sth else wrong earlier. Fixed it.

**Pull request merged**